plugins/input/bitbucket/cloud/cloud.go

```diff
@@ -0,0 +1,163 @@
+	kindSplit := strings.SplitN(cloud.kind, "/", 2)
+	kindType := kindSplit[0]
+	kindKey := kindSplit[1]
+
+	accountID := os.Getenv(cloud.accountID)
+
+	switch kindType {
+	case "project":
+		/*
+		The go-bitbucket SDK does not cover all API endpoints for some reason
+		Will have to get list of projects and search for the project key
+		*/
+
+		// Get list of projects user has access to
+		result, err := cloud.api.Teams.Projects(accountID)
+		if err != nil {
+			fmt.Printf("Authenticate Project: %v\n", err.Error())
+			return false, err
+		}
+
+		// Check if mentioned project is in the list
+		passed := false
+		values := result.(map[string]interface{})["values"].([]interface{})
+		for _, v := range values {
+			passed = kindKey == v.(map[string]interface{})["key"] 
+			if passed {
+				return passed, nil
+			}
+		}
+		return false, fmt.Errorf("Project key not found: %s", kindKey)
+	
+	case "user":
+		// Check if current user can access the repos of the user mentioned (kindKey) in config
+		_, err := cloud.api.Teams.Repositories(kindKey)
+		if err != nil {
+			fmt.Printf("Authenticate User: %v\n", err.Error())
+			return false, err
+		}
+		return true, nil
+	
+	default:
+		// Mentioned kind is unsupported
+		return false, fmt.Errorf("Unsupported kind: %v", kindType)
+	}
+}
+
+// Repositories queries the API and returns a list of repositories mentioned by the kind
+func (cloud Cloud) Repositories(metadata bool) ([]common.Repository, error) {
+	kindSplit := strings.SplitN(cloud.kind, "/", 2)
+	kindType := kindSplit[0]
+	kindKey := kindSplit[1]
+
+	accountID := os.Getenv(cloud.accountID)
+
+	switch kindType {
+	case "project":
+		ro := bitbucket.RepositoriesOptions{ Owner: accountID, Role: "member" }
+		result, err := cloud.api.Repositories.ListForAccount(&ro)
+		if err != nil {
+			fmt.Printf("Get project repositories: %v", err.Error())
+			return nil, err
+		}
+
+		repositories := []common.Repository{}
+		for _, repo := range result.Items {
+			if repo.Project.Key == kindKey {
+
+				repoBytes, _ := json.MarshalIndent(repo, "", "  ")
+				repoJSON := string(repoBytes)
+				httpCloneLink := gjson.Get(repoJSON, `Links.clone.#(name%"http*").href`)
+				slug := gjson.Get(repoJSON, `Slug`)
+
+				newRepo := common.Repository{
+					Slug: slug.String(),
+					Source: httpCloneLink.String(),
+				}
+				repositories = append(repositories, newRepo)
+			}
+		}
+		return repositories, nil
+	
+	case "user":
+		ro := bitbucket.RepositoriesOptions{ Owner: kindKey, Role: "member" }
+		result, err := cloud.api.Repositories.ListForAccount(&ro)
+		if err != nil {
+			fmt.Printf("Get repositories: %v", err.Error())
+			return nil, err
+		}
+
+		repositories := []common.Repository{}
+		for _, repo := range result.Items {
+			newRepo := common.Repository{
+				Slug: repo.Slug,
+			}
+			repositories = append(repositories, newRepo)
+		}
+		return repositories, nil
+	
+	default:
+		return nil, fmt.Errorf("Unsupported kind: %v", kindType)
```

Comment:
Nice.  This is similar to the "default" case statement that I made earlier in this review.